docs/disable-if-parameter.adoc

```diff
@@ -0,0 +1,83 @@
+The annotation is only supported on test method level for parameterized tests.
+Unlike the `@Disabled` API provided in JUnit Jupiter, which disables the test on first encounter of the annotation , `@DisableIfParameter` is validated before each parameterized test execution.
+As a consequence, instead of disabling the entire set of parameterized tests, each test can be evaluated and possibly disabled individually.
+
+[source,java]
+----
+// disable invocations whose parameter contains "she"
+@DisableIfParameter(contains = "she")
+@ParameterizedTest
+@ValueSource(strings = {
+        "Tread lightly, she is near",
+        "Under the snow,",
+        "Speak gently, she can hear",
+        "The daisies grow."
+})
+void interceptContains(String line) {
+}
+----
+
+The test `interceptContains` ordinarily would run four times, but because the first and third lines contain the word "she", those invocations get disabled.
+
+You can specify more than one substring at a time:
+
+[source,java]
+----
+@DisableIfParameter(contains = { "bright", "dust" })
+@ParameterizedTest
+@CsvSource(delimiter = ';', value = {
+        "All her bright golden hair;Tarnished with rust,",
+        "She that was young and fair;Fallen to dust."
+})
+void interceptContainsAny(String line, String line2) {
+}
+----
+
+`DisableIfParameter` will disable the test invocation if any parameter value matches any value from `contains`.
+In this case, the test `interceptContainsAny` should have run two times, but both invocations got disabled.
+The first invocation because the first parameter contains "bright".
+The second invocation because the second parameter contains "dust".
+
+If substrings are not powerful enough, you can also use regular expressions:
+
+[source,java]
+----
+// disable invocations whose parameter ends with 'knew' or 'grew'
+@DisableIfParameter(matches = { ".*knew", ".*grew" })
+@ParameterizedTest
+@ValueSource(strings = {
+        "Lily-like, white as snow,",
+        "She hardly knew",
+        "She was a woman, so",
+        "Sweetly she grew"
+})
+void interceptMatches(String value) {
+}
+----
+
+The second invocation has a parameter that ends with "knew", so it gets disabled.
+The fourth invocation has a parameter that ends with "grew", so it gets disabled.
+
+Just like with `contains`, if any parameter value matches any expression from `matches`, the invocation gets disabled.
+
+You can use both, in which case a test is disabled if any parameter value contains a substring _or_ matches an expression:
+
+[source,java]
+----
+@ParameterizedTest
+@DisableIfParameter(
+        contains = { "sonnet", "life" },
+        matches = "^.*(.+)\\1.*$")
+@ValueSource(strings = {
+        "Peace, Peace, she cannot hear",
+        "Lyre or sonnet,",
+        "All my life’s buried here,",
+        "Heap earth upon it."
+})
+void interceptBoth(String value) {
+}
+----
```

Comment:
To align this tests with the others, and because the regex is not so easy to read like the ones above I would suggest to also add an explanation, which tests are executed and which are not.